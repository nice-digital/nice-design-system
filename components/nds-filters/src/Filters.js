export { FilterPanel } from "./FilterPanel";
export { FilterGroup } from "./FilterGroup";
