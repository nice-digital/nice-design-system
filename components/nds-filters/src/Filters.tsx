export { FilterPanel } from "./FilterPanel";
export { FilterGroup } from "./FilterGroup";
export { FilterOption } from "./FilterOption";
export { FilterSummary, FilterSummaryProps } from "./FilterSummary";
export { FilterByInput } from "./FilterByInput";
