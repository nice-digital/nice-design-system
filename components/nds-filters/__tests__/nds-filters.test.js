"use strict";

import Filters from "../src/Filters";

describe("@nice-digital/nds-filters", () => {
	test.todo("add filter component tests");
});
