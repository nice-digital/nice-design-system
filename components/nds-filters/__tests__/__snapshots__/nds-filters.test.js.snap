// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@nice-digital/nds-filters Crash test should match snapshot 1`] = `
Array [
  <FilterSummary
    activeFilters={
      Array [
        Object {
          "label": "Another filter",
          "onClick": [MockFunction],
        },
      ]
    }
    sorting={
      Array [
        Object {
          "active": true,
          "label": "Active",
        },
        Object {
          "label": "Button",
          "onClick": [MockFunction],
        },
      ]
    }
  >
    <div
      className="filter-summary"
    >
      <div
        className="filter-summary__count"
      >
        <h2
          className="h5 mv--0"
        >
          Showing results 1 to 10 of 1209
        </h2>
      </div>
      <ResultsSorting
        sorting={
          Array [
            Object {
              "active": true,
              "label": "Active",
            },
            Object {
              "label": "Button",
              "onClick": [MockFunction],
            },
          ]
        }
      >
        <div
          className="filter-summary__sorting"
        >
          <label
            className="filter-summary__sort-label"
            htmlFor="sorting-options"
          >
            Sort by
          </label>
          <select
            className="filter-summary__select"
            defaultValue=""
            id="sorting-options"
            onChange={[Function]}
          >
            <option
              key="0"
            >
              Active
            </option>
            <option
              key="1"
            >
              Button
            </option>
          </select>
        </div>
      </ResultsSorting>
      <ResultsFilters
        filters={
          Array [
            Object {
              "label": "Another filter",
              "onClick": [MockFunction],
            },
          ]
        }
      >
        <ul
          aria-label="Active filters"
          className="filter-summary__filters hide-print"
        >
          <li
            className="filter-summary__filter"
            key="Another filter"
          >
            <Tag
              outline={true}
            >
              <span
                className="tag tag--outline"
              >
                Another filter
                <button
                  aria-label="Remove Another filter"
                  className="tag__remove"
                  onClick={[MockFunction]}
                >
                  <SvgRemove>
                    <svg
                      aria-hidden={true}
                      className="icon"
                      fill="none"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M480 392.508c0 10.052-3.52 18.598-10.559 25.637l-51.296 51.296C411.106 476.48 402.56 480 392.508 480c-10.053 0-18.598-3.52-25.638-10.559L256 358.571l-110.87 110.87C138.09 476.48 129.545 480 119.492 480c-10.052 0-18.598-3.52-25.637-10.559l-51.296-51.296C35.52 411.106 32 402.56 32 392.508c0-10.053 3.52-18.598 10.56-25.638L153.428 256 42.559 145.13C35.52 138.09 32 129.545 32 119.492c0-10.052 3.52-18.598 10.56-25.637l51.295-51.296C100.894 35.52 109.44 32 119.492 32c10.053 0 18.598 3.52 25.638 10.56L256 153.428l110.87-110.87C373.91 35.52 382.455 32 392.508 32c10.052 0 18.598 3.52 25.637 10.56l51.296 51.295C476.48 100.894 480 109.44 480 119.492c0 10.053-3.52 18.598-10.559 25.638L358.571 256l110.87 110.87C476.48 373.91 480 382.455 480 392.508z"
                        fill="currentColor"
                      />
                    </svg>
                  </SvgRemove>
                  <span
                    className="visually-hidden"
                  >
                    Remove 
                    Another filter
                     filter
                  </span>
                </button>
              </span>
            </Tag>
          </li>
        </ul>
      </ResultsFilters>
    </div>
  </FilterSummary>,
  <FilterPanel
    fallback={
      Object {
        "action": "/submit",
        "method": "GET",
      }
    }
    heading="Some filters"
    onSubmit={[MockFunction]}
  >
    <form
      action="/submit"
      method="GET"
      onSubmit={[MockFunction]}
    >
      <div
        className="filter-panel"
      >
        <h2
          className="filter-panel__heading"
        >
          <button
            aria-controls="filter-panel-body"
            aria-expanded={true}
            onClick={[Function]}
            type="button"
          >
            <SvgChevronDown
              className="filter-panel__heading-icon filter-panel__heading-icon--expanded"
            >
              <svg
                aria-hidden={true}
                className="filter-panel__heading-icon filter-panel__heading-icon--expanded"
                fill="none"
                focusable="false"
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
                  fill="currentColor"
                />
              </svg>
            </SvgChevronDown>
            Some filters
          </button>
        </h2>
        <div
          aria-hidden={false}
          className="filter-panel__body"
          id="filter-panel-body"
        >
          <FilterGroup
            heading="Some filters"
            headingLevel={3}
            key=".0"
          >
            <div
              className="filter-group"
            >
              <h3
                className="filter-group__heading"
              >
                <button
                  aria-controls="group-some-filters"
                  aria-expanded={true}
                  onClick={[Function]}
                  type="button"
                >
                  <SvgChevronDown
                    className="filter-group__heading-icon filter-group__heading-icon--expanded"
                  >
                    <svg
                      aria-hidden={true}
                      className="filter-group__heading-icon filter-group__heading-icon--expanded"
                      fill="none"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
                        fill="currentColor"
                      />
                    </svg>
                  </SvgChevronDown>
                  <span
                    id="group-heading-some-filters"
                  >
                    Some filters
                     
                  </span>
                </button>
              </h3>
              <fieldset
                aria-hidden={false}
                className="filter-group__options"
                id="group-some-filters"
              >
                <legend>
                  Some filters
                </legend>
                <FilterOption
                  groupHeading="Some filters"
                  groupId="some-filters"
                  isSelected={true}
                  key=".0"
                  onChanged={[MockFunction]}
                >
                  <label
                    className="filter-option"
                    htmlFor="filter_some-filters_a-filter"
                  >
                    <input
                      checked={true}
                      id="filter_some-filters_a-filter"
                      name="some-filters"
                      onChange={[Function]}
                      title="Some filters - A filter"
                      type="checkbox"
                      value="A filter"
                    />
                    <span
                      className="filter-option__text"
                    >
                      A filter
                    </span>
                  </label>
                </FilterOption>
              </fieldset>
            </div>
          </FilterGroup>
          <FilterByInput
            headingLevel={3}
            key=".1"
            label="Filter by name"
            name="Filter by name"
          >
            <div
              className="inputFilterBox"
            >
              <h3
                className="inputFilterBox__heading"
              >
                <button
                  aria-controls="inputFilter-Filter by name"
                  aria-expanded={true}
                  onClick={[Function]}
                  type="button"
                >
                  <SvgChevronDown
                    className="filter-group__heading-icon filter-group__heading-icon--expanded"
                  >
                    <svg
                      aria-hidden={true}
                      className="filter-group__heading-icon filter-group__heading-icon--expanded"
                      fill="none"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
                        fill="currentColor"
                      />
                    </svg>
                  </SvgChevronDown>
                  Filter by name
                </button>
              </h3>
              <div
                aria-hidden={false}
                className="inputFilterBox__controls"
                id="inputFilter-Filter by name"
              >
                <Input
                  label="Filter by name"
                  name="Filter by name"
                  type="text"
                >
                  <div
                    className="input"
                  >
                    <label
                      className="input__label"
                      htmlFor="Filter by name"
                    >
                      Filter by name
                    </label>
                    <input
                      className="input__input"
                      id="Filter by name"
                      name="Filter by name"
                      type="text"
                    />
                  </div>
                </Input>
                <Button
                  className="ml--0 mb--0"
                  type="submit"
                >
                  <button
                    className="btn ml--0 mb--0"
                    type="submit"
                  >
                    Filter
                  </button>
                </Button>
              </div>
            </div>
          </FilterByInput>
        </div>
      </div>
    </form>
  </FilterPanel>,
]
`;
